### DISCUSSION TITLE
container/container-network

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#204] [General]
# container/container-network

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/container/container-network.html

**Comment by pangzitudou** [comment]
"当 Node1 中的 Container-1 与 Node2 中的 Container-2 通信时，根据上面的路由规则，交由 flannel.1 接口处理，flannel.1 的职责是作为 VXLAN 网络的 VETH 端点负责封包/解包。“ 中的VETH 应该是VTEP吧

**Reply by isno** [reply]
谢谢，已修改。

**Comment by JackCharles** [comment]
”那的地 MAC 地址呢？也就是 Node2 中 flannel.1 设备的 MAC 地址怎么获取？“

那的地 => 那目的地

**Reply by isno** [reply]
谢谢，我内容里面还是有不少笔误的地方，我也在逐一校验修改。

**Comment by madeinchinalmc** [comment]
Underlay 底层网络模式它在k8s里面的使用场景建议给个例子，这里就科普了下知识感觉稍微有点突兀

**Reply by isno** [reply]
很早之前，docker 用 macvlan 实现组网。但 k8s 下面的 underlay 组网，我也没见过。所以，这里只是简单提一下，有这么个模式

**Comment by madeinchinalmc** [comment]
上面记录的意思是，目的 MAC 地址为 92:8d:c4:85:16:ad（ Node2 VTEP 设备的 MAC 地址）的数据帧封装后，应该发往哪个目的IP（宿主机网络中节点的 IP）。根据上面的记录可以看出，Node1 中 UDP 包的目的 IP 应该为 192.168.2.103，也就是 Node2 节点的 IP。

这里图文不符了，目的MAC地址按照你的配图应该是：ba:74:f9:db:69:c1，目的ip按照你的配图应该是192.168.50.3

**Comment by madeinchinalmc** [comment]
三层路由模式章节，这条路由规则的含义是，目的地属于 10.244.1.3/24 网段的 IP 包应被送往 cni0 网桥，这里和你配图也不符合，应该是”目的地属于 100.10.0.0/24网段的 IP 包...“

**Comment by zjyyzhengli** [comment]
1. “凡是发往 100.10.2.0/24 网段的 IP 报文，都需要经过接口 flannel.1，并且最后一跳的网关地址是 **10.224.1.0**（也就是 Node2 中 VTEP 的设备）。” -- 这个网关地址是啥？是不是写错了？
2. “Linux 内核中默认为 VXLAN 分配的 UDP 监听端口为 8472。因此目的地端口即 8472。” -- 这个端口不是**4789**吗？

**Comment by Alohalt** [comment]
图文的ip好像很多对不上
